### PR TITLE
MVP & Stretch completed

### PR DESCRIPTION
**Pull request by ajb85**
Completed so far:
Initial commit
this.js

### CODE REVIEW

**Comment by ajb85** [comment]
prototypes.js & MVP done

**Comment by ajb85** [comment]
Stretch complete

File: assignments/prototypes.js [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
-  
+
 /*
   === GameObject ===
   * createdAt
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
 */
-
+function GameObject(charAttrs) {
```
Comment (GannonDetroit) [inline assignments/prototypes.js:23]: good job

File: assignments/prototypes.js [hunk 2]
```diff
@@ -41,70 +72,137 @@
...
-  console.log(swordsman.healthPoints); // 15
-  console.log(mage.name); // Bruce
-  console.log(swordsman.team); // The Round Table
-  console.log(mage.weapons); // Staff of Shamalama
-  console.log(archer.language); // Elvish
-  console.log(archer.greet()); // Lilith offers a greeting in Elvish.
-  console.log(mage.takeDamage()); // Bruce took damage.
-  console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+console.log(mage.createdAt); // Today's date
```
Comment (GannonDetroit) [inline assignments/prototypes.js:117]: great job! everything looks good on my end. You did a fantastic job

File: assignments/prototypes.js [hunk 3]
```diff
@@ -41,70 +72,137 @@
...
+Hero.prototype = Object.create(Humanoid.prototype);
+Hero.prototype.shieldBash = function(target, damage) {
+  if (this.weapons.includes("Shield")) {
+    target.healthPoints -= damage;
+    console.log(
+      `${this.name} did ${damage} damage to ${target.name} with Shield Bash`
+    );
+  } else {
+    console.log(
+      `${this.name} tried to Shield Bash but forgot he doesn't have a shield.`
```
Comment (GannonDetroit) [inline assignments/prototypes.js:145]: lol I love this one. (I use to play WoW and I loved being a Paladin so shield slam like abilities are a personal favorite)

File: assignments/prototypes.js [hunk 4]
```diff
@@ -41,70 +72,137 @@
...
+
+const hero = new Hero({
+  createdAt: new Date(),
+  dimensions: {
+    length: 1,
+    width: 2,
+    height: 4
+  },
+  healthPoints: 10,
+  name: "Lord Beef Broth",
```
Comment (GannonDetroit) [inline assignments/prototypes.js:178]: lol interesting choice in name

File: assignments/prototypes.js [hunk 5]
```diff
@@ -41,70 +72,137 @@
...
+  target.dimensions.height = Math.floor(target.dimensions.height / 2);
+  console.log(`${this.name} shrunk ${target.name} to half his height!`);
+};
 
-  // Stretch task: 
-  // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
-  // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+hero.shieldBash(villain, 2);
```
Comment (GannonDetroit) [inline assignments/prototypes.js:205]: great job on finishing and making the stretch problem work!

File: assignments/this.js [hunk 6]
```diff
@@ -1,26 +1,57 @@
...
-* 4. 
+* 1. When a function 'this' occupies is applied to the global scope, its value will be the window object (no me gusta)
+* 2. When using a method, the value of 'this' will be the object preceding the .method()
+* 3. When used in a constructor function, the value of 'this' will be the object created & returned by the constructor
+* 4. .bind(), .call(), and .apply() specifically define 'this' so its value will always be the object supplied
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
```
Comment (GannonDetroit) [inline assignments/this.js:12]: good job

File: assignments/this.js [hunk 7]
```diff
@@ -1,26 +1,57 @@
...
-
+myFunc();
+function myFunc() {
+  //console.log("Example 1: ", this); // mayhem, uncomment if you would like mayhem
```
Comment (GannonDetroit) [inline assignments/this.js:15]: yes... much mayhem

File: assignments/this.js [hunk 8]
```diff
@@ -1,26 +1,57 @@
...
+}
 // code example for Window Binding
 
 // Principle 2
```
Comment (GannonDetroit) [inline assignments/this.js:19]: good job

File: assignments/this.js [hunk 9]
```diff
@@ -1,26 +1,57 @@
...
+  greeting: "Hello!",
+  speak: function() {
+    console.log(`Example 2: Good day, sir.  To you I say: ${this.greeting}`);
+    console.log("Implicit: ", this);
+  }
+};
+formalGreetings.speak();
 // code example for Implicit Binding
 
 // Principle 3
```
Comment (GannonDetroit) [inline assignments/this.js:30]: good job

File: assignments/this.js [hunk 10]
```diff
@@ -1,26 +1,57 @@
...
+      `Example 3: What's up, man?  I'd like to say: ${this.greeting}`
+    );
+    console.log("Constructor: ", this);
+  };
+}
+const informal = new InformalGreeting();
+informal.speak();
 // code example for New Binding
 
 // Principle 4
```
Comment (GannonDetroit) [inline assignments/this.js:44]: good job

**Review by GannonDetroit — COMMENTED** [review]
great job. You nailed the MVP and the stretch and it looks like you had some fun while doing it. 

keep it up!

**Pull request merged**